sphinx/ext/intersphinx.py

```diff
@@ -288,6 +288,41 @@ def missing_reference(app, env, node, contnode):
         if not objtypes:
             return
         objtypes = ['%s:%s' % (domain, objtype) for objtype in objtypes]
+
+        if domain == 'std' and node['reftype'] == 'option':
```

Comment:
I think this is responsibility of domains. Since 1.6, we provides `Domain#get_full_qualified_name()` to normalize target name on each domain. Can you migrate this to the method?